### ISSUE TITLE
References which depend on "mscorlib, Version=2.0.5.0... packages\FluentAssertions.4.0.0\lib\net45\FluentAssertions.Core.dll

### ISSUE THREAD

**Issue by KarlZ** [dennisdoomen/fluentassertions#299]
When I build I get an warning: Found conflicts between versions of the same dependent assembly that could not be resolved. When I set verbosity to detailed, I get:
 There was a conflict between "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" and "mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes".
3>      "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" was chosen because it was primary and "mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes" was not.
3>      References which depend on "mscorlib, Version=4.0.0.0,<snip>
...
FluentAssertions.dll...
References which depend on "mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes" [].
3>          C:\TFS\Projects\Main\Source\WebApp\packages\FluentAssertions.4.0.0\lib\net45\FluentAssertions.Core.dll

**Comment by michael-fischer** [comment]
I'm pretty sure this happens because they target silver light so fluentAssertions.core is a PCL (haven't looked) which will link to version 2.0.5.0.  I thought a binding redirect for the 3 files (System, System.Core, mscorlib) it complains about would fix it but did not have any luck.

**Comment by dennisdoomen** [comment]
Maybe related to this? http://www.paraesthesia.com/archive/2013/01/21/using-portable-class-libraries-update-net-framework.aspx/

**Comment by jfoshee** [comment]
This seems related to targeting to .NET 4.6, but I'm not sure what's going on yet.

**Comment by dennisdoomen** [comment]
Can you create a little project to reproduce this?

**Comment by michael-fischer** [comment]
Interestingly enough, when I try to create a demo project.  I still get the conflicts in the build output (with the verbosity turned up) but I don't get the warnings in the error dialog. I don't know if @KarlZ would have more success.

Michael Fischer

**Comment by dennisdoomen** [comment]
Still, would you be able to attach the ZIP or push it to some repo?

**Comment by dennisdoomen** [comment]
@KarlZ what framework is your project targetting?

**Comment by KarlZ** [comment]
@dennisdoomen They all target `.NET Framework 4.6`

**Comment by dennisdoomen** [comment]
Could you please try 4.1? It contains some slight changes workaround to some bugs in the nuget framework resolution rules.

**Comment by KarlZ** [comment]
Hmmm... More investigation required. I tried it yesterday and got the same issue on our "real" project which is sizable. So I was putting together a sample - and of course it didn't happen. So I'll need some time to see if I can figure out if there is some caching on the other project etc. I'll let you know when I have something conclusive.

**Comment by dennisdoomen** [comment]
Good luck investigating...

**Comment by KarlZ** [comment]
Ok... We had one test project with this reference:
`<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />`
While the others had this:
`<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />`
When I changed that one to match all the others, the error went away.
Version 10.1.0.0 was in `C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\ReferenceAssemblies\v2.0\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll` while version 10.0.0.0 was in `C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\PublicAssemblies\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll`. At this point I'm not sure why the error messages pointed at `FluentAssertions.Core`. I would expect a newer version (10.1 instead of 10.0) for the UnitTestFramework to work better?
At any rate, now our project builds correctly with no conflicts.
Just as a check, I went back to the code from several days ago where we were using `FluentAssertions 4.0.1.0` and changed the UnitTestFramework reference. The error went away on that. I just wanted to make sure that both the upgrade to 4.1 and changing the test framework reference to 10.0.0.0 weren't required.

Thanks!
